backend/de.metas.business/src/main/java/de/metas/bpartner/effective/BPartnerEffectiveBL.java

```diff
@@ -0,0 +1,242 @@
+		bPartnerBuilder.poInvoiceRule(getEffectiveValue(
+				bPartnerRecord,
+				I_C_BPartner::getPO_InvoiceRule,
+				InvoiceRule::ofNullableCode,
+				this::getDefaultInvoiceRule)
+		);
+
+		//noinspection DataFlowIssue
+		bPartnerBuilder.paymentRule(getEffectiveValue(
+				bPartnerRecord,
+				I_C_BPartner::getPaymentRule,
+				PaymentRule::ofCode,
+				() -> null)
+		);
+
+		//noinspection DataFlowIssue
+		bPartnerBuilder.poPaymentRule(getEffectiveValue(
+				bPartnerRecord,
+				I_C_BPartner::getPaymentRulePO,
+				PaymentRule::ofCode,
+				() -> null)
+		);
+
+		return bPartnerBuilder.build();
+	}
+
+	@Nullable
+	private I_C_BP_Group getParentGroup(@Nullable final I_C_BP_Group bpGroup)
+	{
+		if (bpGroup == null)
+		{
+			return null;
+		}
+		final BPGroupId parentGroupId = BPGroupId.ofRepoIdOrNull(bpGroup.getParent_BP_Group_ID());
+		return parentGroupId != null ? bpGroupDAO.getById(parentGroupId) : null;
+	}
+
+
+
+	@Nullable
+	private <T, V> T getEffectiveValue(
+			@NonNull final I_C_BPartner bPartner,
+			@NonNull final I_C_BP_Group bpGroup,
+			@Nullable final I_C_BP_Group bpParentGroup,
+			@NonNull final Function<I_C_BPartner, V> bPartnerValueExtractor,
+			@NonNull final Function<I_C_BP_Group, V> bpGroupValueExtractor,
+			@NonNull final Function<V, T> valueMapper,
+			@NonNull final Supplier<T> defaultValueSupplier)
+	{
+		return CoalesceUtil.coalesceSuppliers(
+				() -> {
+					final V value = bPartnerValueExtractor.apply(bPartner);
+					return value != null ? valueMapper.apply(value) : null;
+				},
+				() -> {
+					final V value = bpGroupValueExtractor.apply(bpGroup);
+					return value != null ? valueMapper.apply(value) : null;
+				},
+				() -> {
+					if (bpParentGroup == null)
+					{
+						return null;
+					}
+					final V value = bpGroupValueExtractor.apply(bpParentGroup);
+					return value != null ? valueMapper.apply(value) : null;
+				},
+				defaultValueSupplier
+		);
+	}
+
+	@Nullable
+	private <T, V> T getEffectiveValue(
+			@NonNull final I_C_BPartner bPartner,
+			@NonNull final Function<I_C_BPartner, V> bPartnerValueExtractor,
+			@NonNull final Function<V, T> valueMapper,
+			@NonNull final Supplier<T> defaultValueSupplier)
+	{
+		return CoalesceUtil.coalesceSuppliers(
+				() -> {
+					final V value = bPartnerValueExtractor.apply(bPartner);
+					return value != null ? valueMapper.apply(value) : null;
+				},
+				defaultValueSupplier
+		);
+	}
+
+
+	@Nullable
+	private PricingSystemId getDefaultSalesPricingSystemId(@NonNull final OrgId orgId)
+	{
+		if (orgId.isRegular())
+		{
+			final OrgInfo orgInfo = orgDAO.getOrgInfoById(orgId);
+			return orgInfo.getPricingSystemId();
+		}
+		return null;
+	}
+
+	@NonNull
+	private InvoiceRule getDefaultInvoiceRule()
+	{
+		final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+
+		final String invoiceRule = sysConfigBL.getValue(DEFAULT_INVOICE_RULE, X_C_Order.INVOICERULE_AfterDelivery);
+		if (Check.isNotBlank(invoiceRule))
```

Comment:
pls use StringUtils#trimBlankToNull or StringUtils#trimBlankToOptional to be on the safe side

Reply:
e.g. your code will fail if by mistake the user forgets a space before or after the sysconfig value...